commit fa7726a4d9b91e0a8fb6fbbd819cd20f116f6c21
Author: Justin Stitt <justinstitt@google.com>
Date:   Thu Sep 21 03:51:04 2023 +0000

    i3c: replace deprecated strncpy
    
    `strncpy` is deprecated for use on NUL-terminated destination strings [1].
    
    We should prefer more robust and less ambiguous string interfaces.
    
    We expect adap->name to be NUL-terminated based on i2c_adapter name use:
    | dev_dbg(&adap->dev, "adapter [%s] registered\n", adap->name);
    
    NUL-padding does not seem to be required as `master` is zero-allocated
    and `i3c_master_to_i2c_adapter` simply returns a field from within
    `master`:
    |       master = devm_kzalloc(dev, sizeof(*master), GFP_KERNEL);
    ...
    |       struct i2c_adapter *adap = i3c_master_to_i2c_adapter(master);
    ...
    |       static struct i2c_adapter *
    |       i3c_master_to_i2c_adapter(struct i3c_master_controller *master)
    |       {
    |               return &master->i2c;
    |       }
    
    This means that `adap->name` should already be filled with NUL-bytes.
    
    Considering the above, a suitable replacement is `strscpy` [2] due to
    the fact that it guarantees NUL-termination on the destination buffer
    without unnecessarily NUL-padding.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20230921-strncpy-drivers-i3c-master-c-v1-1-9fdb8d8169e1@google.com
    Signed-off-by: Alexandre Belloni <alexandre.belloni@bootlin.com>

diff --git a/drivers/i3c/master.c b/drivers/i3c/master.c
index 87283e4a4607..8573ca507708 100644
--- a/drivers/i3c/master.c
+++ b/drivers/i3c/master.c
@@ -2309,7 +2309,7 @@ static int i3c_master_i2c_adapter_init(struct i3c_master_controller *master)
 	adap->dev.parent = master->dev.parent;
 	adap->owner = master->dev.parent->driver->owner;
 	adap->algo = &i3c_master_i2c_algo;
-	strncpy(adap->name, dev_name(master->dev.parent), sizeof(adap->name));
+	strscpy(adap->name, dev_name(master->dev.parent), sizeof(adap->name));
 
 	/* FIXME: Should we allow i3c masters to override these values? */
 	adap->timeout = 1000;